src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs

```diff
@@ -683,7 +683,7 @@ static void addERR_ParamsCantBeWithModifier(BindingDiagnosticBag diagnostics, Sy
                 // error CS0721: '{0}': static types cannot be used as parameters
                 diagnostics.Add(
                     ErrorFacts.GetStaticClassParameterCode(parameter.ContainingSymbol.ContainingType?.IsInterfaceType() ?? false),
-                    owner.Locations.IsEmpty ? parameterSyntax.GetLocation() : owner.Locations[0],
+                    parameterSyntax.Type!.Location,
```

Comment:
We probably can't assume that `Type` is not null, as it should be possible to force a lambda parameter without an explicitly-specified `Type` to a `static` type through type inference in generics. We should fall back to the whole `parameterSyntax` if there's no type.

Reply:
6fc77cfaba50e60c2d46200b9ba7d9d82ac446dc